blocks/article-grid/article-grid.js

```diff
@@ -0,0 +1,364 @@
+import { getLibs, buildBlock } from '../../scripts/utils.js';
+import initBanner from '../banner/banner.js';
+
+const miloLibs = getLibs();
+const defaultLimit = 9;
+const ROOT_MARGIN = 20;
+
+// Used for getting localized text
+const replacePlaceholder = async (key) => {
+  const { replaceKey } = await import(`${miloLibs}/features/placeholders.js`);
+  const { getConfig } = await import(`${miloLibs}/utils/utils.js`);
+  const result = await replaceKey(key, getConfig());
+
+  return result;
+};
+
+// get data
+function getMetadataTags(doc) {
+  const metaTagElements = doc.head.querySelectorAll('meta[property="article:tag"]');
+  const tagContents = Array.from(metaTagElements, (metaTag) => metaTag.getAttribute('content')).filter(Boolean);
+
+  return JSON.stringify(tagContents);
+}
+
+async function getArticleDetails(articleLink) {
+  const { getMetadata } = await import(`${miloLibs}/utils/utils.js`);
+
+  const path = new URL(articleLink).pathname;
+  const resp = await fetch(path);
+  if (!resp || !resp.ok) {
+    // eslint-disable-next-line no-console
+    console.log(`Could not retrieve metadata for ${path}`);
+    return null;
+  }
+  const html = await resp.text();
+  const parser = new DOMParser();
+  const doc = parser.parseFromString(html, 'text/html');
+
+  const trimEndings = ['|Adobe', '| Adobe', '| Adobe Blog', '|Adobe Blog'];
+  let title = getMetadata('og:title', doc).trim();
+  const ending = trimEndings.find((el) => title.endsWith(el));
+  [title] = title.split(ending);
+
+  const image = doc.querySelector('img');
+
+  // same data format for decorating media block
+  return {
+    title,
+    description: getMetadata('description', doc),
+    image: image.src,
+    imageAlt: image.alt ? image.alt : title,
+    tags: getMetadataTags(doc),
+    path,
+  };
+}
+
+async function fetchArticleFeedData(blogIndex) {
+  const { getConfig } = await import(`${miloLibs}/utils/utils.js`);
+  const { updateLinkWithLangRoot } = await import(`${miloLibs}/utils/helpers.js`);
+  const { feed, articles, limit } = blogIndex.config;
+
+  // fetch article data based on links
+  if (articles) {
+    const fetchArticleDataPromise = articles.map((article) => getArticleDetails(article));
+    let articleData = [];
+
+    articleData = await Promise.all(fetchArticleDataPromise);
```

Comment:
Can `articleData` be a const?
```suggestion
    const articleData = await Promise.all(fetchArticleDataPromise);
```

Reply:
updated, thanks!